File: paper/src/main/java/de/s42/paper/PaperMessageService.java
package de.s42.paper;

import de.s42.core.MessageService;
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
import org.bukkit.entity.Player;

public class PaperMessageService implements MessageService {
    private static final LegacyComponentSerializer legacyComponentSerializer = LegacyComponentSerializer.legacy('&');

    @Override
    public void sendActionBar(Player player, String message) {
        player.sendActionBar(legacyComponentSerializer.deserialize(message));
    }
}

File: paper/src/main/kotlin/tr/s42/paper/PaperMessageService.kt
package tr.s42.paper

import tr.s42.core.MessageService
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import org.bukkit.entity.Player

class PaperMessageService : MessageService {

    companion object {
        private val legacyComponentSerializer = LegacyComponentSerializer.legacy('&')
    }

    override fun sendActionBar(player: Player, message: String) {
        player.sendActionBar(legacyComponentSerializer.deserialize(message))
    }
}